apps/web/src/pages/templates/editor/TemplateEditorPage.tsx

```diff
@@ -78,6 +79,14 @@ export default function TemplateEditorPage() {
     }
   }, [template]);
 
+  useEffect(() => {
+    if (environment && template) {
+      if (environment._id !== (template as any)._environmentId) {
```

Comment:
```suggestion
      if (environment._id !== (template as any)._environmentId && template._parentId) {
```

I think adding this check can help avoid strange situations in the future